### ISSUE TITLE
startsWith / endsWith for numbers

### ISSUE THREAD

**Issue by juulderuysscher** [balderdashy/sails-mongo#244]
In the code now only the 'endsWith' and 'startsWith' is implemented for strings. But a solution to get all documents which starts or ends with a certain number is:

`{ "$where": "/^.*<number>$/i.test(this.<fieldName>)"}`

Would be nice if also support for numbers can be added.

I tried implementing this but my implementation didn't work. I added in the /lib/query/index.js file:
```js
            if(_.isNumber(val)) {
                // Handle `endsWith` for number
                if(modifier === 'endsWith') {
                    val = parseInt(val);
                    val =  '.*' + val;
                    obj['$where'] = (new RegExp('^' + val + '$', 'i')) + '.test(this.' + field + ')';
                    return obj;
                }
            }
```

**Comment by sailsbot** [comment]
Thanks for posting, @juulderuysscher.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
